### ISSUE TITLE
Prediction of 3D boxes format

### ISSUE THREAD

**Issue by ofir1080** [facebookresearch/omni3d#13]
Hi!
First of all, thank you very much for this awesome work :)
I have been trying to understand in detail the prediction format of the boxes.
in the paper it is said:
![image](https://user-images.githubusercontent.com/40369687/195943539-2d80b2d3-d1be-4eca-8a40-032e2956c648.png)

1. does [u,v] represent the pixels where the projected box center is? that is simply (K x center_cam)?
2. the virtual depth params are set to f_v = H_v = 512, which derives z_v = H / f ? is f given?
3. are the per-category w0,h0.l0 are given? could  not find where...
4. about the allocentric rotation, when I convert p to egocentric rotation, does this correspond to 'R_cam'?

Thank you very much for the time and sorry all the questions

**Comment by longbowzhang** [comment]
Actually I have a similar question as (3) in yours. Then I dive a little bit deep in the codes and find the following
https://github.com/facebookresearch/omni3d/blob/47580fd372d14b4ededf4a3c040b92df5aa49aab/cubercnn/modeling/roi_heads/roi_heads.py#L115-L118
It looks like those per-category means are learned.

**Comment by ofir1080** [comment]
Thanks
however, in the paper:
![image](https://user-images.githubusercontent.com/40369687/195983253-a0e37caa-cbf4-42fe-a755-b55e08cbeafa.png)
How do you interpret that?

**Comment by gkioxari** [comment]
Hi all! 

Let me answer your questions, and expand a bit for pedagogical reasons! 

> Per-category $(w_0, h_0, l_0)$ means.

As you have noticed we have support for a few settings. We treat the $(w_0, h_0, l_0)$ as parameters, initialized fro scratch or from *training* ground truth information, which can be finetuned or frozen. In our default setting, the means $(w_0, h_0, l_0)$ are pre-computed from the training ground truth (see [here](
https://github.com/facebookresearch/omni3d/blob/main/cubercnn/util/math_util.py#L292)) and are not further learned (aka frozen) (see [here](https://github.com/facebookresearch/omni3d/blob/47580fd372d14b4ededf4a3c040b92df5aa49aab/cubercnn/modeling/roi_heads/roi_heads.py#L461-L468)). **Interestingly, we found that learning them from scratch was just as good but would slow down convergence and model training, as you'd expect!**

> Is $(u,v)$ in pixel space?

Yes. $(u, v)$ is the projected 3D object center on the image plane. So if $(X,Y,Z)$ is the object's 3D center, $(u,v)$ is its projection on the image plane using the camera matrix $K$.

> Virtual depth

The focal length for each image is given. The focal length is a camera parameter that provides information about scale. The focal length $f$ varies per image of course. So what virtual depth does, is that it projects the real (world) space to a scale-invariant space which is **much friendlier** for neural nets to cast predictions in. 

> Allocentric vs Egocentric

That's right. Poses are stored in egocentric format and can be converted in allocentric ones (see our appendix). We predict allocentric poses.

**Comment by Sondosmohamed1** [comment]
@gkioxari  Thank you very much for your work and  clarification, i have question regarding the camera rotation and location is the model is affected with camera rotation and location (external matrix ), because i generated custom dataset  from different cameras ,and i rotated the cameras and put them in different locations , then for every camera and object  i generated the R_cam from roll , pitch ,yaw matrices with respect to camera ( the difference between the camera rotation and the object rotation) in every axis  , my question is what i did is right , or this rotation of the camera i made  will affect the predication
